### ISSUE TITLE
Adding extra CMAKE_BUILD_TYPE?

### ISSUE THREAD

**Issue by mathomp4** [ecmwf/ecbuild#33]
A query for the "canonical" ecbuild. We (in GEOS) have a fork of ecbuild that we use in our project. Recently [I had to patch our fork](https://github.com/GEOS-ESM/ecbuild/pull/3) because I wanted to add a new `CMAKE_BUILD_TYPE` and it wasn't one of the allowed types by ecbuild:
https://github.com/ecmwf/ecbuild/blob/d4347cf92466c742ef989444b2acbf7c867a338e/cmake/ecbuild_define_build_types.cmake#L51-L59

My question is: is there a better way to do this? I added `Aggressive` as I was adding "aggressive" flags, but say someone wanted to use the more "official" `MinSizeRel` (which is [in the CMake source](https://github.com/Kitware/CMake/blob/62816ff88c6d1bd8d1c00de5ed448ee915d69e00/Modules/Compiler/GNU-Fortran.cmake#L21)). As far as I can tell, they couldn't with ecbuild.

**Comment by mathomp4** [comment]
@tlmquintino Just a ping. Do you have a good method of a way to "add" additional `CMAKE_BUILD_TYPES` to an ecbuild system (extra functions, toolchains, etc.)?

**Comment by mathomp4** [comment]
@marcosbento As you seem to be a big name in this repo, do you think ECMWF would be adverse to an additional CMake Build Type if I made a PR?

I just did a check today and our model can now build perfectly with the latest ecbuild (develop at least) but we need the `Aggressive` build type.

**Comment by mathomp4** [comment]
Note: If not, no worries as that's what forks are for, but, well, always nice to be able to use the mother repo if needed/desired. :)

**Comment by wdeconinck** [comment]
Hi @mathomp4 I'm thinking that if we would just remove the assertion when not using one of the supported build types, it could also work.

**Comment by mathomp4** [comment]
@wdeconinck Oh...yeah, that might be better/easier. 🙂 

Or perhaps, issue an `ecbuild_warning` or `ecbuild_info` that a user has selected an unrecognized `CMAKE_BUILD_TYPE` (to ecbuild) and the flags to use are on them?

**Comment by wdeconinck** [comment]
Indeed, if you can live with a ecbuild_warning in your configuration with Aggressive build type that is fine for me.

**Comment by mathomp4** [comment]
@wdeconinck I will do some testing and I'll ask our (few) users of our Aggressive flags if they mind a CMake warning. But then we already have a few anyway so it'd just probably be noise. 😄
